providers/azure/azure_service.go

```diff
@@ -15,9 +15,19 @@
 package azure
 
 import (
+	"github.com/Azure/go-autorest/autorest"
 	"github.com/GoogleCloudPlatform/terraformer/terraformutils"
+	"github.com/hashicorp/go-azure-helpers/authentication"
 )
 
 type AzureService struct { //nolint
 	terraformutils.Service
 }
+
+func (s AzureService) GetSubscriptionID() string {
+	return s.Args["config"].(authentication.Config).SubscriptionID
+}
+
+func (s AzureService) GetAuthorizer() autorest.Authorizer {
+	return s.Args["authorizer"].(autorest.Authorizer)
+}
```

Comment:
@rotemavni How did all the other azure services get around without these methods?

Reply:
@nimrodkor It's hard coded in all other azure services, I'm planning to refactor all of them to use these methods

Reply:
@rotemavni Let's do it in 2 steps then, to make the contribution easier:
For now - do it like all other services. We'll merge it and contribute it
Then refactor across all Azure services - and merge + contribute it

WDYT?

Reply:
@nimrodkor It's ok, I removed this refactor